lib/checkbool.cpp

```diff
@@ -429,8 +430,8 @@ void CheckBool::pointerArithBoolError(const Token *tok)
     reportError(tok,
                 Severity::error,
                 "pointerArithBool",
-                "Converting pointer arithmetic result to bool. The bool is always true unless there is undefined behaviour.\n"
-                "Converting pointer arithmetic result to bool. The boolean result is always true unless there is pointer arithmetic overflow, and overflow is undefined behaviour. Probably a dereference is forgotten.", CWE571, false);
+                "Converting pointer arithmetic result to bool. The bool is always true unless there is undefined behavior.\n"
+                "Converting pointer arithmetic result to bool. The boolean result is always true unless there is pointer arithmetic overflow, and overflow is undefined behavior. Probably a dereference is forgotten.", CWE571, false);
```

Comment:
Ok. This weekend or Monday I'll revert back using the original word. No
problem at all.

On 13 August 2016 at 11:39, PKEuS <notifications@github.com> wrote:

> In lib/checkbool.cpp
> <https://github.com/danmar/cppcheck/pull/819#discussion_r74685302>:
>
> > @@ -429,8 +430,8 @@ void CheckBool::pointerArithBoolError(const Token *tok)
> >      reportError(tok,
> >                  Severity::error,
> >                  "pointerArithBool",
> > -                "Converting pointer arithmetic result to bool. The bool is always true unless there is undefined behaviour.\n"
> > -                "Converting pointer arithmetic result to bool. The boolean result is always true unless there is pointer arithmetic overflow, and overflow is undefined behaviour. Probably a dereference is forgotten.", CWE571, false);
> > +                "Converting pointer arithmetic result to bool. The bool is always true unless there is undefined behavior.\n"
> > +                "Converting pointer arithmetic result to bool. The boolean result is always true unless there is pointer arithmetic overflow, and overflow is undefined behavior. Probably a dereference is forgotten.", CWE571, false);
>
> I would prefer to keep the message as it is. Anyway, if the message is
> changed, the test suite needs to be changed, too, otherwise there are test
> failures.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/danmar/cppcheck/pull/819/files/b974b4da8925304429a314a943c3ce11c98be615#r74685302>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAgYnqJFyIkld_HK9BV0v3smNxolm-3Iks5qfZ7-gaJpZM4JjE_6>
> .
>



-- 
Roberto Martelloni
boos @ http://boos.core-dumped.info